lib/Reader/readerir.cpp

```diff
@@ -283,6 +282,10 @@ void GenIR::readerPrePass(uint8_t *Buffer, uint32_t NumBytes) {
       ABIMethodSignature(MethodSignature, *this, *JitContext->TheABIInfo);
   Function = ABIMethodSig.createFunction(*this, *JitContext->CurrentModule);
 
+  if (JitContext->Options->DoInsertStatepoints) {
+    GcFuncInfo = JitContext->GcInfo->newGcInfo(Function);
+  }
+
```

Comment:
It seems to me it would be cleaner to just always create a GcFuncInfo. Then you could remove a bunch of conditional logic below....